src/Adapter/Product/CommandHandler/BulkUpdateProductStatusHandler.php

```diff
@@ -66,7 +76,28 @@ public function handle(BulkUpdateProductStatusCommand $command): void
      */
     protected function handleSingleAction(ProductId $productId, $command = null)
     {
-        $this->productStatusUpdater->updateStatus($productId, $command->getNewStatus());
+        $product = $this->productMultiShopRepository->getByShopConstraint($productId, $command->getShopConstraint());
+        $wasVisibleOnSearch = $this->productIndexationUpdater->isVisibleOnSearch($product);
+        $wasActive = (bool) $product->active;
```

Comment:
I think here we can already check if status is the expected one and avoid going further.
```
if ((bool) $product->active === $command->getNewStatus) {
    return;
}
```

Reply:
Actually I don't think we can because the command is multi shop, so if the constraint targets several shop we still need to force the updated on other shops regardless of the current value (which is only the value from default shop).
Or I can add this optimization but only in single shop context

Reply:
But I'm not sure it's worth it it would add a three conditions if statement for a minimalist optimization that would complexity the readability of the code at best, and would cause bugs at worst So I' rather not change this part if you don't mind